base/libgit2/callbacks.jl

```diff
@@ -224,7 +229,7 @@ function credentials_callback(libgit2credptr::Ptr{Ptr{Void}}, url_ptr::Cstring,
 
     if isset(allowed_types, Cuint(Consts.CREDTYPE_USERPASS_PLAINTEXT))
         defaultcreds = reset!(UserPasswordCredentials(true), -1)
-        credid = "$schema$host"
+        credid = "$(isempty(schema) ? "ssh" : schema)://$host"
```

Comment:
We could just assign "ssh" to be the schema when one is missing but this would then modify the `git_url` output from `git@github.com` to `ssh://git@github.com` which would no longer resemble the URL passed in by the user.